File: src/SOP8/Egovframework/Base/base-history-ibll/src/main/java/egovframework/base/history/ibll/request/RequestExcelSOPPartialHistory.java
package egovframework.base.history.ibll.request;

import java.util.ArrayList;
import java.util.List;

public class RequestExcelSOPPartialHistory {

    private int beginYear = -1;
    private int beginMonth = -1;
    private int beginDay = -1;
    private int endYear = -1;
    private int endMonth = -1;
    private int endDay = -1;

    private String disasterCategoryName = null; // null이면 전체
    private String actionStepName = null;       // null이면 전체
    private List<Integer> actionStepHistoryNos = new ArrayList<>();

    public int getBeginYear() {
        return beginYear;
    }

    public void setBeginYear(int beginYear) {
        this.beginYear = beginYear;
    }

    public int getBeginMonth() {
        return beginMonth;
    }

    public void setBeginMonth(int beginMonth) {
        this.beginMonth = beginMonth;
    }

    public int getBeginDay() {
        return beginDay;
    }

    public void setBeginDay(int beginDay) {
        this.beginDay = beginDay;
    }

    public int getEndYear() {
        return endYear;
    }

    public void setEndYear(int endYear) {
        this.endYear = endYear;
    }

    public int getEndMonth() {
        return endMonth;
    }

    public void setEndMonth(int endMonth) {
        this.endMonth = endMonth;
    }

    public int getEndDay() {
        return endDay;
    }

    public void setEndDay(int endDay) {
        this.endDay = endDay;
    }

    public String getDisasterCategoryName() {
        return disasterCategoryName;
    }

    public void setDisasterCategoryName(String disasterCategoryName) {
        this.disasterCategoryName = disasterCategoryName;
    }

    public String getActionStepName() {
        return actionStepName;
    }

    public void setActionStepName(String actionStepName) {
        this.actionStepName = actionStepName;
    }

    public List<Integer> getActionStepHistoryNos() {
        return actionStepHistoryNos;
    }

    public void setActionStepHistoryNos(List<Integer> actionStepHistoryNos) {
        this.actionStepHistoryNos = actionStepHistoryNos;
    }
}

File: src/SOP8/Base/History/Base.History.IBLL/Request/RequestExcelSOPPartialHistory.cs
using System.Collections.Generic;

namespace Base.History.IBLL.Request
{
    public class RequestExcelSOPPartialHistory
    {
        private int m_nBeginYear = -1;
        private int m_nBeginMonth = -1;
        private int m_nBeginDay = -1;
        private int m_nEndYear = -1;
        private int m_nEndMonth = -1;
        private int m_nEndDay = -1;
        private string m_strDisasterCategoryName = null;
        private string m_strActionStepName = null;
        private List<int> m_actionStepHistoryNos = new List<int>();
        private string m_strSubjectName = null;

        public int BeginYear
        {
            get { return m_nBeginYear; }
            set { m_nBeginYear = value; }
        }

        public int BeginMonth
        {
            get { return m_nBeginMonth; }
            set { m_nBeginMonth = value; }
        }

        public int BeginDay
        {
            get { return m_nBeginDay; }
            set { m_nBeginDay = value; }
        }

        public int EndYear
        {
            get { return m_nEndYear; }
            set { m_nEndYear = value; }
        }

        public int EndMonth
        {
            get { return m_nEndMonth; }
            set { m_nEndMonth = value; }
        }

        public int EndDay
        {
            get { return m_nEndDay; }
            set { m_nEndDay = value; }
        }

        // null이면 전체
        public string DisasterCategoryName
        {
            get { return m_strDisasterCategoryName; }
            set { m_strDisasterCategoryName = value; }
        }

        // null이면 전체
        public string ActionStepName
        {
            get { return m_strActionStepName; }
            set { m_strActionStepName = value; }
        }

        public List<int> ActionStepHistoryNos
        {
            get { return m_actionStepHistoryNos; }
            set { m_actionStepHistoryNos = value; }
        }

        public string SubjectName
        {
            get { return m_strSubjectName; }
            set { m_strSubjectName = value; }
        }
    }
}
